openapi.yml

```diff
@@ -31,81 +54,149 @@ paths:
         sellAmount:
+          description: "Amount of sellToken to be sold in atoms"
           $ref: "#/components/schemas/TokenAmount"
-        sellToken:
-          $ref: "#/components/schemas/Address"
-        sellTokenTip:
+        buyAmount:
+          description: "Amount of buyToken to be bought in atoms"
           $ref: "#/components/schemas/TokenAmount"
-        orderType:
-          description: Is this a buy order or sell order?
-          type: string
-          enum: [buy, sell]
-        fillType:
-          description: Is this a fill-or-kill order a partially fillable order?
-          type: string
-          enum: [fillorkill, partial]
-        nonce:
-          $ref: "#/components/schemas/Nonce"
         validTo:
-          description: Time offset in seconds from epoch until which the order is valid. uint32.
+          description: Unix timestamp until the order is valid. uint32.
+          type: integer
+        appData:
+          description: |
+            Arbitrary identifier sent along with the order.
+            Could be used to track the interface or other meta-aspects of the order. uint32 encoded
           type: integer
+        tip:
+          description: "Tip to compensate solvers in atoms"
+          $ref: "#/components/schemas/TokenAmount"
+        orderType:
+          description: "The type is either a buy or sell order"
+          $ref: "#/components/schemas/OrderType"
+        partiallyFillable:
+          description: Is this a fill-or-kill order or a partially fillable order?
+          type: boolean
         signature:
           description: 65 bytes encoded as hex without `0x` prefix. v + r + s from the spec.
           example: "0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
-    OrderExtra:
-      description: Extra order data that is returned to users when querying orders but not provided by users when creating orders.
+    OrderMetaData:
+      description: |
+        Extra order data that is returned to users when querying orders 
+        but not provided by users when creating orders.
       type: object
       properties:
         creationTime:
           description: Creation time of the order. Encoded as ISO 8601 UTC.
           type: string
-          example: "2020-10-22T07:57:18Z"
+          example: "2020-12-03T18:35:18.814523Z"
         owner:
           $ref: "#/components/schemas/Address"
-    OrderFull:
+        UID:
+          $ref: "#/components/schemas/UID"
+    Order:
       allOf:
-        - $ref: "#/components/schemas/OrderNew"
-        - $ref: "#/components/schemas/OrderExtra"
+        - $ref: "#/components/schemas/OrderCreation"
+        - $ref: "#/components/schemas/OrderMetadata"
+    UID:
+     description: |
+      Unique identifier for the order: orderDigest_ownerAddress_validTo,
+      where orderDigest = keccak256(orderStruct). bytes32.
+     type: string
```

Comment:
This isn't bytes32. Should probably say something like "65 bytes encoded as hex without `0x`. Bytes 0 to 30 are the order digest, bytes 30 to 42 the owner address and bytes 42..66 valid to. (i made up the sizes).

Reply:
`s/65/56/`. (`0..32 = orderDigest`, `32..52 = ownerAddress` and `52..56 = validTo`) but otherwise agree.